### PR TITLE
Use IOU/Expense reportID for IOU actions

### PR DESCRIPTION
**Pull request by luacmartins**
Add the IOU reportAction to the IOU/Expense report instead of the parent chatReport.

We are gonna start creating IOU reportActions directly in the IOU/Expense report and as part of the switch, we are temporarily skipping test so that Auth integration tests pass. The plan to make the switch is:

1. Merge https://github.com/Expensify/Web-Expensify/pull/37375 and CP to prod
2. Web-E Integrations tests in this https://github.com/Expensify/Auth/pull/7863 should pass
3. Merge the https://github.com/Expensify/Auth/pull/7863
4. Deploy the Auth PR
5. Merge https://github.com/Expensify/Web-Expensify/pull/37337 that re-enables tests using the correct reportID
6. CP the https://github.com/Expensify/Web-Expensify/pull/37337 PR to prod
7. Merge https://github.com/Expensify/App/pull/18604, CPing it to staging to minimize the amount of time IOUs are broken
8. Run the bulkCQ from https://github.com/Expensify/Expensify/issues/282420 to change the reportID of all previous IOU actions

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/281893

### Tests
In dev, pull these [Auth](https://github.com/Expensify/Auth/pull/7863) and [Web-E](https://github.com/Expensify/Web-Expensify/pull/37337) branches

Pre-requisites: 2 accounts you have access to

**Request money**
1. In accountA, go offline
2. Request money from a user you haven't chatted with before, `+ > Request money`
3. Verify:
  - A new chat report is created between you and accountB. The chat should be greyed out (pending state)
  - A new iou report is created between you and accountB and is displayed in the LHN (this report has a different header showing the request total). The report should be greyed out (pending state)
  - In the iou report, you see the IOU request preview. The preview should be greyed out (pending state).
4. Go online and verify that all pending states are resolved, i.e. the reports and preview are no longer greyed out
5. From the chat report, request money again to the same user.
6. Verify that a new preview is added to the IOU report.

**Delete request**
8. Tap on the request preview and delete the request
9. Verify that a `deleted the $X request` message is added to the IOU report

**Pay money request**
10. In accountB, navigate to the IOU report with accountA
11. Tap on the pay button and settle elsewhere
12. Verify that a `settled up $X` message is added to the IOU report

**Workspace case**
1. In account A, create a new Workspace `+ > New workspace`
2. Navigate to the Workspace chat
3. Repeat the steps for Request money, Delete request and Pay money request above.

- [x] Verify that no errors appear in the JS console

### Offline tests
Covered above

### QA Steps
Same as above.

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/22219519/c691d0e4-67be-4c7f-bb4d-1626f7738dfd

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/22219519/9af2a988-66b9-4a81-89c8-725cb9d21966

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by aldo-expensify** [comment]
What is the progress on this?

File: src/libs/actions/IOU.js [hunk 1]
```diff
@@ -660,28 +765,37 @@ function deleteMoneyRequest(chatReportID, iouReportID, moneyRequestAction, shoul
...
+    const currentUserEmail = optimisticIOUAction.actorEmail;
+    let updatedIOUReport = {};
+    if (ReportUtils.isExpenseReport(iouReportID)) {
+        updatedIOUReport = {...iouReport};
+
+        // Because of the Expense reports are stored as negative values, we add the total from the amount
+        updatedIOUReport.total += amount;
+    } else {
+        updatedIOUReport = IOUUtils.updateIOUOwnerAndTotal(
+            iouReports[`${ONYXKEYS.COLLECTION.REPORT}${iouReportID}`],
```
Comment (mountiny) [inline src/libs/actions/IOU.js:777]: You got the iouReport defined above
```suggestion
            iouReport,
```

**Review by mountiny — COMMENTED** [review]
When deleting the IOU preview is shown in place of the deleted message 
image

That one exist in the IOU report LHN preview and in the RHP but the chat repord does not show it correctly

Settled up elsewhere worked but it showed as $0.00 optimistically first 
image


This should be the expense report with the Ferrari workspace 
image

It seems like the name and icon of the policy are not being loaded correctly 

Unable to delete a expense on the workspace and getting this error:
`Throwing exception with message: '404 Account not found, possible bad authToken?' f`

File: src/libs/actions/IOU.js [hunk 2]
```diff
@@ -707,24 +815,16 @@ function deleteMoneyRequest(chatReportID, iouReportID, moneyRequestAction, shoul
     const failureData = [
         {
             onyxMethod: Onyx.METHOD.MERGE,
-            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReportID}`,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${iouReportID}`,
             value: {
-                [optimisticReportAction.reportActionID]: {
+                [optimisticIOUAction.reportActionID]: {
+                    pendingAction: null,
```
Comment (s77rt) [inline src/libs/actions/IOU.js:821]: Do we actually want to clear the `pendingAction` here? Asking because last time we wanted to keep the pendingAction to have the 50% opacity. https://github.com/Expensify/App/blob/293f23b57be2bca3f3eb0af903d5e0c2894cb36a/src/components/OfflineWithFeedback.js#L88.

Old discussion: https://github.com/Expensify/App/pull/17583#discussion_r1174318102

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/libs/actions/IOU.js [hunk 3]
```diff
@@ -68,165 +230,63 @@ function requestMoney(report, amount, currency, payeeEmail, participant, comment
...
+    let iouReport;
+
+    if (!isNewIOUReport) {
         if (isPolicyExpenseChat) {
-            moneyRequestReport = {...iouReports[`${ONYXKEYS.COLLECTION.REPORT}${chatReport.iouReportID}`]};
+            iouReport = {...iouReports[`${ONYXKEYS.COLLECTION.REPORT}${chatReport.iouReportID}`]};
 
             // Because of the Expense reports are stored as negative values, we substract the total from the amount
-            moneyRequestReport.total = ReportUtils.isExpenseReport(moneyRequestReport) ? moneyRequestReport.total - amount : moneyRequestReport.total + amount;
+            iouReport.total -= amount;
```
[the remark this answers is not in the document]
Comment (luacmartins) [reply src/libs/actions/IOU.js:249]: Correct. Only Expense reports live in `policyExpenseChats` and we already check for that in the previous conditional.

File: src/libs/actions/IOU.js [hunk 4]
```diff
@@ -235,11 +295,11 @@ function requestMoney(report, amount, currency, payeeEmail, participant, comment
...
             currency,
             comment: parsedComment,
-            iouReportID: moneyRequestReport.reportID,
+            iouReportID: iouReport.reportID,
             chatReportID: chatReport.reportID,
             transactionID: optimisticTransaction.transactionID,
-            reportActionID: optimisticReportAction.reportActionID,
-            createdReportActionID: isNewChat ? optimisticCreatedAction.reportActionID : 0,
+            reportActionID: optimisticIOUAction.reportActionID,
+            createdReportActionID: isNewChatReport ? optimisticCreatedActionForChat.reportActionID : 0,
```
[the remark this answers is not in the document]
Comment (luacmartins) [reply src/libs/actions/IOU.js:302]: For this one, we're still passing the created reportAction for the Chat report. We'll need to update the API command to take both created actions eventually.

**Comment by luacmartins** [comment]
Updated! At this point we can refactor `splitBill` to also use `buildOnyxDataForMoneyRequest`, since the logic is very similar. I'll do that in a follow up though.

File: src/libs/actions/IOU.js [hunk 5]
```diff
@@ -235,11 +318,11 @@ function requestMoney(report, amount, currency, payeeEmail, participant, comment
...
```
Comment (mountiny) [inline src/libs/actions/IOU.js:325]: @luacmartins I think we need to pass the created report action for the IOU report too. I am getting two created report actions here 
image

image


We wont be rendering the Created report actions so this problem technically wont eb visible so is that the intention?

File: src/libs/actions/IOU.js [hunk 6]
```diff
@@ -616,23 +667,33 @@ function createSplitsAndOnyxData(participants, currentUserLogin, amount, comment
...
                 onyxMethod: Onyx.METHOD.MERGE,
-                key: `${ONYXKEYS.COLLECTION.REPORT}${oneOnOneChatReport.reportID}`,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${oneOnOneIOUReport.reportID}`,
                 value: {
-                    hasOutstandingIOU: existingOneOnOneChatReport ? existingOneOnOneChatReport.hasOutstandingIOU : false,
-                    iouReportID: existingOneOnOneChatReport ? existingOneOnOneChatReport.iouReportID : null,
-                    errorFields: {
-                        createChat: {
-                            [DateUtils.getMicroseconds()]: Localize.translateLocal('report.genericCreateReportFailureMessage'),
+                    [oneOnOneIOUAction.reportActionID]: {
```
Comment (s77rt) [inline src/libs/actions/IOU.js:689]: Shouldn't this be `oneOnOneCreatedActionForIOU` instead of `oneOnOneIOUAction`? (if not `existingOneOnOneIOUReport`)

Or maybe it's the other way around? https://github.com/Expensify/App/blob/85e7e5b9017566fcaa4819aec11935cb55045011/src/libs/actions/IOU.js#L217
here we should use `iouAction` instead of `iouCreatedAction` because I see this one is already missing the `isNewIOUReport` condition.


Either way let's be consist here about where the error message should be at.

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/libs/actions/IOU.js (hunk 5 above)
Comment (mountiny) [reply src/libs/actions/IOU.js:336]: Ideally will be handled here https://github.com/Expensify/App/pull/18775#pullrequestreview-1425452759

Comment (luacmartins) [reply src/libs/actions/IOU.js]: Updated to pass the param. I'll update Auth and Web-E too

**Comment by luacmartins** [comment]
Updated!

**Review by mountiny — APPROVED** [review]

**Review by s77rt — COMMENTED** [review]
Code looks good.
Let me know when this is ready to test.

**Comment by mountiny** [comment]
Thanks @s77rt it will hopefully be early on Monday

**Comment by luacmartins** [comment]
thanks for the reviews @s77rt @mountiny!

**Comment by mountiny** [comment]
@luacmartins lint failing

**Comment by luacmartins** [comment]
@mountiny resolved

**Comment by luacmartins** [comment]
@s77rt this is ready to be tested in staging. FYI we have an issue with the call to `OpenReport` it'll be in an infinite loop right now because it doesn't return data for reports of type = iou or expense. This will probably make testing challenging.

**Comment by mountiny** [comment]
I think there might be some new places which need updating the chat report/ iou report, I will try to see if I can find them

**Review by Julesssss — DISMISSED** [review]
Testing well for me against [Cristi's Web-E PR](https://github.com/Expensify/Web-Expensify/pull/37438). other than settling up from the IOU, but I think that's a separate issue .

**Review by mountiny — CHANGES_REQUESTED** [review]
@luacmartins we need to change this to iouReport/ expnese report https://github.com/Expensify/App/blob/7ad2be02a9e7c9dc19e8a6556d3c8efde5ba5721/src/components/ReportActionItem/MoneyRequestAction.js#L115

**Comment by luacmartins** [comment]
Updated

**Comment by luacmartins** [comment]
@s77rt that should be fixed soon as we are in the midst of deploying those changes.

**Comment by s77rt** [comment]
Clicking IOU preview opens a new report instead of RHN IOU details

https://github.com/Expensify/App/assets/16493223/abe407c3-ef28-483a-b84d-2376c4db2dbf

**Comment by mountiny** [comment]
> Clicking IOU preview opens a new report instead of RHN IOU details

This is not a bug but feature. Its a transaction thread, you can chat with your friend/ manager about that particular request there. We are adding the update header there

**Review by mountiny — APPROVED** [review]

**Comment by s77rt** [comment]
How am I supposed to delete a money request?

**Comment by s77rt** [comment]
Ah I got it

**Review by Julesssss — APPROVED** [review]

**Comment by s77rt** [comment]
Requesting money from workspace fails


https://github.com/Expensify/App/assets/16493223/e48f3a4c-f20e-4c23-bda9-8c1459395c6c


https://github.com/Expensify/App/assets/16493223/b3809c86-3067-4a6d-9933-7a71e1ff2837

**Comment by luacmartins** [comment]
@s77rt yes, we are aware and working onti

**Comment by s77rt** [comment]
@luacmartins Is it a blocker or can we merge and fix that in a follow up PR?

**Comment by luacmartins** [comment]
@s77rt the issue is in the API, so we can continue here

**Comment by s77rt** [comment]
Other two minor issues:
1. The avatar in LHN always shows the current user's avatar
2. There is an option to request money from IOU report but it's broken (requesting money from chat report works well)

**Comment by luacmartins** [comment]
@s77rt both of those are known and being worked on

1. https://expensify.slack.com/archives/C01GTK53T8Q/p1684254793691709
2. https://github.com/Expensify/App/pull/19033

**Comment by s77rt** [comment]
@luacmartins Okay one more :grin:
- You can't delete the last existing money request
- The IOU preview for deleted request is still shown


https://github.com/Expensify/App/assets/16493223/b8f5dfae-81bb-40d8-87fb-ef3aa44b4998

**Comment by luacmartins** [comment]
I'm not able to reproduce the delete error. What error are you getting back from the API? 

As for the IOUPReview, it should still be shown.

**Comment by s77rt** [comment]
Error 401

```js
{
    "jsonCode": 401,
    "message": "401 Unauthorized report NVP.",
    "onyxData": [
        {
            "onyxMethod": "merge",
            "key": "reportActions_6792924156649370",
            "value": {
                "errors": {
                    "1684260166419151": "Auth DeleteMoneyRequest returned an error"
                }
            }
        },
        {
            "onyxMethod": "merge",
            "key": "reportActions_6792924156649370",
            "value": {
                "errors": {
                    "1684260166419151": "Auth DeleteMoneyRequest returned an error"
                }
            }
        }
    ],
    "requestID": "7c8588974f9c9a17-SJC"
}
```

**Comment by luacmartins** [comment]
@s77rt that seems unrelated to the changes in this PR. We already have another internal issue to track that error.

**Comment by s77rt** [comment]
## Reviewer Checklist

- [X] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [X] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [X] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



https://github.com/Expensify/App/assets/16493223/c1a05f8b-b6c5-4970-a735-4e6b18e5d779


</details>

<details>
<summary>Mobile Web - Chrome</summary>


mweb-chrome

</details>

<details>
<summary>Mobile Web - Safari</summary>


mweb-safari

</details>

<details>
<summary>Desktop</summary>



https://github.com/Expensify/App/assets/16493223/7c51fd8c-f797-484a-a58b-0513f07818a3


</details>

<details>
<summary>iOS</summary>


https://github.com/Expensify/App/assets/16493223/77bb1f87-2fd7-4693-bc5d-7839e492351e

</details>

<details>
<summary>Android</summary>


android

</details>

**Comment by s77rt** [comment]
@luacmartins Checklist is ready. Just waiting for a confirmation for the reported bug if it's something we should fix in this PR.

**Review by s77rt — APPROVED** [review]
LGTM! :rocket:

**Comment by luacmartins** [comment]
Thanks for the thorough review @s77rt

File: src/components/ReportActionItem/MoneyRequestAction.js [hunk 7]
```diff
@@ -112,7 +112,7 @@ const MoneyRequestAction = (props) => {
                 undefined,
                 CONST.REPORT.NOTIFICATION_PREFERENCE.ALWAYS,
                 props.action.reportActionID,
-                props.chatReportID, // Needs to be changed to iouReportID
+                props.requestReportID,
```
Comment (s77rt) [inline src/components/ReportActionItem/MoneyRequestAction.js:115]: @luacmartins Can you please explain why was this needed? We think this is causing this bug https://github.com/Expensify/App/issues/19487

Comment (luacmartins) [reply src/components/ReportActionItem/MoneyRequestAction.js:115]: We should not be building a thread when clicking the preview on a Split  action. We instead show a split details page that is being worked on [here](https://github.com/Expensify/App/pull/19390)

Comment (mountiny) [reply src/components/ReportActionItem/MoneyRequestAction.js:115]: it has to be for other cases, we should probably consider treating this a bit differntly for the split case, need to explore more there.
